src/index.tsx

```diff
@@ -1,16 +1,55 @@
 import React from "react";
 import ReactDOM from "react-dom";
+import {
+  createBrowserRouter,
+  RouterProvider,
+  redirect
+} from "react-router-dom";
 import "./index.css";
 import App from "./App";
 import logo from "./logo.svg";
 
+const router = createBrowserRouter([
+  {
+    path: "/",
+    element: <App />,
+    loader: ({ params }) => {
+      // If the app was accessed through old URL format, redirect
+      const txhash = new URLSearchParams(window.location.search).get("t");
+
+      // This loader is executed when accessing tx/:txhash, we need this check to avoid infinite loop
+      if (txhash?.length === 66 && !params.txhash) {
+        return redirect(`tx/${txhash}`);
+      }
+    },
+    children: [
+      {
+        path: "tx",
+        element: null,
+        loader: ({ params }) => {
+          const { txhash } = params;
+          if (!txhash || txhash.length !== 66) {
+            return redirect("/");
+          }
+        },
+        children: [
+          {
+            path: "/tx/:txhash",
```

Comment:
Are we sure we require the preceeding slash here or it should be `tx/:txhash`?

Reply:
Yes, I tested it, we need to have the path of the parent here, if we have only `tx/:txhash` or `:txhash` it won't work